code/clue.js

```diff
@@ -1,59 +1,350 @@
+  image: "assets/mustard.png",
+  occupation: "butcher",
+  favouriteWeapon: "poison",
+};
+
+const mrsWhite = {
+  firstName: "Ana",
+  lastName: "White",
+  color: "white",
+  description: "She is very weak",
+  age: 28,
+  image: "assets/white.png",
+  occupation: "painter",
+  favouriteWeapon: "axe",
+};
 
 // CREATE OBJECTS FOR ALL THE WEAPONS, ADD MORE CHARACTERISTICS TO THE WEAPONS IF YOU LIKE.
 
 const rope = {
-  name: 'rope',
-  weight: 10
-}
+  name: "rope",
+  weight: 10,
+  id: "rope",
+};
+
+const knife = {
+  name: "knife",
+  weight: 5,
+  id: "knife",
+};
+
+const candlestick = {
+  name: "candlestick",
+  weight: 7,
+  id: "candlestick",
+};
+
+const dumbbell = {
+  name: "dumbbell",
+  weight: 30,
+  id: "dumbbell",
+};
+
+const poison = {
+  name: "poison",
+  weight: 2,
+  id: "poison",
+};
+
+const axe = {
+  name: "axe",
+  weight: 12,
+  id: "axe",
+};
+
+const bat = {
+  name: "bat",
+  weight: 13,
+  id: "bat",
+};
+
+const trophy = {
+  name: "trophy",
+  weight: 20,
+  id: "trophy",
+};
+
+const pistol = {
+  name: "pistol",
+  weight: 90,
+  id: "pistol",
+};
 
 // THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
 
+console.log(colonelMustard.color);
+console.log(pistol.weight);
+
 // NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
 
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
-]
+  professorPlum,
+  missScarlet,
+  mrsPeackok,
+  colonelMustard,
+  mrsWhite,
+];
+
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol,
+];
+
+const rooms = [
+  "Dining Room",
+  "Conservatory",
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio",
+];
 
-const weapons = []
+console.log(suspects);
+console.log(suspects[3].occupation);
+console.log(weapons);
+console.log(rooms.indexOf("Spa"));
 
-const rooms = []
+const weaponRooms = {
+  rope: ["Conservatory", "Hall", "Theater", "Patio", "Observatory"],
+  knife: ["Kitchen"],
+  candlestick: [
+    "Dining Room",
+    "Conservatory",
+    "Study",
+    "Library",
+    "Lounge",
+    "Ballroom",
+    "Hall",
+    "Spa",
+    "Living Room",
+    "Guest House",
+    "Patio",
+    "Theater",
+  ],
+  dumbbell: ["Guest House"],
+  poison: [
+    "Dining Room",
+    "Conservatory",
+    "Kitchen",
+    "Living Room",
+    "Guest House",
+    "Patio",
+  ],
+  axe: ["Conservatory", "Billiard Room", "Hall", "Patio"],
+  bat: ["Billiard Room", "Guest House"],
+  trophy: ["Living Room", "Guest House"],
+  pistol: ["Living Room", "Gest House"],
+};
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
 // YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+
+const randomSelector = (array) => {
```

Comment:
When there is only one argument in the function, there is a convention not to use parenthesis around the argument, in this case `array`. Like this: 

`const randomSelector = array => {  
return array[Math.floor(Math.random() * array.length)];
};`

Reply:
As I can read from the JS documentations, the { brackets } and ( parentheses ) and "return" are optional, but may be required. Can you please share where you have read that convention not to use parenthesis around one argument?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

Also my VSCode by default formats the JS by adding those parentheses if I try to remove them

Reply:
Sandra, regarding your question, this is "longer" way of writing this precise function :

```
const randomSelector = array => {
  return array[Math.floor(Math.random() * array.length)];
};
```

And this is "shorter" version of the same function. By "shorter" I mean that we can write everything in one line if there is only one line inside "longer" version of function. When we create "shorter" version, we can omit two things : 1. brackets `{}` and 2. `return` keyword. JavaScript will add it by itself, so you it's there, but "invisible" :

```
const randomSelector = array => array[Math.floor(Math.random() * array.length)];
```